### PR TITLE
stateManagement using react context api 

### PR DESCRIPTION
**Pull request by Justinnn07**
> 🚨 Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

### Please check if the PR fulfills these requirements

- [x] Make sure you are requesting to **NEW-UI**. Don't request other protected Branches like **staging/master**
- [x] Make sure no conflicts are present in the code, if so please resolve it(_Tip: Always fetch upstream_)
- [x] Your Commit messages should make sense.
- [x] Don't push your package.lock.json as this project uses yarn.lock already.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Describe your changes

- Bought State management(React context API ) to the app .. 

- We can now access the req. data on any component!

❤️ Thank you!

### CODE REVIEW

**Comment by Justinnn07** [comment]
@stephin007 pls review and merge!

File: src/App.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ import "./App.css";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 
 import { ThemeProvider } from "styled-components";
-import { useDarkMode } from "./components/useDarkMode";
-import { lightTheme, darkTheme } from "./components/themes";
-import { GlobalStyles } from "./components/globalStyles";
+import { useDarkMode } from "./Themes/useDarkMode";
+import { lightTheme, darkTheme } from "./Themes/themes";
+import { GlobalStyles } from "./Themes/globalStyles";
```
Comment (stephin007) [inline src/App.js:8]: @Justinnn07 why the name of the folder was changed?

Comment (Justinnn07) [reply src/App.js:8]: > @Justinnn07 why the name of the folder was changed?

Themes are not under components right ? so, thought of making a folder for themes

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -17,27 +17,27 @@ import SearchIcon from "@material-ui/icons/Search";
...
+      districtCode,
+      vaccinePerPage,
+      currentPage,
+    },
+    dispatch,
+  ] = useDataLayerValue();
+
   const [pin, setPin] = useState("");
   const [formattedDate, setFormattedDate] = useState("");
   const [selectedDate, setSelectedDate] = useState(new Date());
```
Comment (stephin007) [inline src/components/Home/Home.js:40]: @Justinnn07 why are the above states not used under context?

Comment (Justinnn07) [reply src/components/Home/Home.js:40]: These states most prob. won't be used outside these components , so thought of maintaining as it is! 

Well, if you want i can specify them under context

File: src/components/Pagination/Pagination.js [hunk 3]
```diff
@@ -1,17 +1,15 @@
 import React from "react";
 import "./Pagination.css";
-import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
-import ChevronRightIcon from "@material-ui/icons/ChevronRight";
```
Comment (stephin007) [inline src/components/Pagination/Pagination.js:3]: these icons are not needed? @Justinnn07

**Comment by stephin007** [comment]
@Justinnn07 Can you also add the api calls to state? i mean right now we are adding only the states!

**Comment by stephin007** [comment]
@Justinnn07 Also Please fill the PR Template!

File: src/components/Pagination/Pagination.js (hunk 3 above)
Comment (Justinnn07) [reply src/components/Pagination/Pagination.js:3]: I am making the UI of the pagination little attractive .. moreover, the prev button was not working tooo!

**Comment by Justinnn07** [comment]
> @Justinnn07 Can you also add the api calls to state? i mean right now we are adding only the states!

Is it necessary ? I am adding every api response to the reducer

**Comment by Justinnn07** [comment]
> @Justinnn07 Also Please fill the PR Template!

sure!

**Comment by stephin007** [comment]
> Is it necessary ? I am adding every api response to the reducer

State management means writing less code right? is this possible what i said?

**Comment by stephin007** [comment]
Also, @Justinnn07 are you removing pagination?

**Comment by Justinnn07** [comment]
> Also, @Justinnn07 are you removing pagination?

Ohk :)

**Comment by Justinnn07** [comment]
> Also, @Justinnn07 are you removing pagination?

No, I will be using Material-ui Pagination

**Comment by stephin007** [comment]
OKay @Justinnn07 it would be better if you raise another PR for pagination changes and use this PR solely for State Management!

please do the necessary

**Comment by Justinnn07** [comment]
> OKay @Justinnn07 it would be better if you raise another PR for pagination changes and use this PR solely for State Management!
> 
> please do the necessary

Sure, will do!

**Pull request closed without merging**